### ISSUE TITLE
Disabling netrw disables the spelling files download suggestion

### ISSUE THREAD

**Issue by doronbehar** [neovim/neovim#7189]
```
$ nvim --version
NVIM v0.2.0
Build type: Release
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.2.0/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui      
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

```
- Vim (version: ) behaves differently? no
- Operating system/version: Arch Linux
- Terminal name/version (`$TERM`): `rxvt-unicode-256color`

### Steps to reproduce

1. Put in `init.vim`:

```
let g:loaded_netrwPlugin = 1
```

Thus prevents loading netrw. In the [help file](https://github.com/neovim/neovim/blob/master/runtime/doc/pi_netrw.txt#L180), it is also suggested to `let g:loaded_netrw = 1` as well but it is not nescesarry to prove the bug.

2. run `nvim` or `vim`:

```
nvim # or vim
```
3. run inside it:
```
set spell
set spelllang=he " Just for example..
```

### Actual behaviour

`{n,}vim` don't suggest to download the spell files from public ftp server of vim.

### Expected behaviour

downloading spell files from vim's public ftp server is suggested although netrw is disabled.

**Comment by justinmk** [comment]
This is kind of a bug+enhancement. We need a `download()` function or something like that, which doesn't depend on netrw. We generally expect `curl` to be available (we ship `curl.exe` in the Windows package), so we could wrap that.

**Comment by doronbehar** [comment]
it seems that almost every function in [`runtime/autoload/spellfile.vim`](https://github.com/neovim/neovim/blob/master/runtime/autoload/spellfile.vim) depends on commands from `Netrw`, I'm working on a draft to fix it.

**Comment by justinmk** [comment]
@doronbehar Might want to look at https://github.com/mattn/webapi-vim/blob/master/autoload/webapi/http.vim#L125 for wrapping curl.

Other option is to ship a Lua library that removes the need for curl. That's probably farther away though.

**Comment by Mattemagikern** [comment]
Is someone working on this? 

I'm, very, new to the project but might be able to make a curl wrapper for the project. 
What is the desired PR?
I'm thinking a C(or possibly a lua + c or just lua if possible) to wrap curl and rewrite the runtime/autoload/spellfile.vim + dependencies to make a complete workaround from netrw. But as I'm new I don't know if thats a viable solution?

**Comment by KillTheMule** [comment]
I wouldn't write such a wrapper in C, and lua is such a nice language...

**Comment by doronbehar** [comment]
Having this implemented in pure Lua seems to me the most elegant solution. There's network support for Lua with https://github.com/diegonehab/luasocket , perhaps this could be bundled in Neovim's Lua? Or is this too much?

**Comment by justinmk** [comment]
nvim 0.4 now exposes all features of libuv so I don't think luasocket is needed . 

    :help vim.loop

**Comment by teto** [comment]
curl or wget are widely available. I would prefer having a lua provider (and all the other providers converted to lua too :D) around these 2 than bundle http-related code in neovim.

**Comment by cuducos** [comment]
I wrote a prototype that is a drop-in replacement for `spellfile.vim` called [`spellfile.nvim`](https://github.com/cuducos/spellfile.nvim) that:
* integrates seamlessly since it uses the same `autocmd` as the native `spellfile.vim`
* has no dependency in `netrw` (it uses `curl` or `wget` for the download)
* is fully written in Lua
* has tests

Is there any chance this can be adapted to become part of Neovim itself? I would love have some guidance and work on that if that's relevant!

**Comment by justinmk** [comment]
@cuducos yes, your plugin looks simple enough to include, PR welcome! Guidance is:

- keep the "surface area" of the plugin close to zero (though if the old Vimscript plugin exposes some public functions/commands, we may need to continue providing those)
- minimize the number of files (the less, the better), e.g. `path.lua` should be eliminated.
- ensure the code matches our codebase conventions (run `stylua` on it)
- delete all files related to old spellfile.vim, where possible.
